### DISCUSSION TITLE
how to effeciently add globlal css

### DISCUSSION THREAD

**Discussion by jpengelbrecht** [nuxt/nuxt#16634] [Questions] [answered]
I have global sass being included in my project and i cant find an efficient way to add it to the project.

there seems to be 2 popular ways to add css in your project.

```
 vite: {
    plugins: [svgLoader()],
    css: {
        preprocessorOptions: {
            scss: {
                additionalData: `
                  @import "~/assets/styles/main.scss";
          `,

            },
        },
    },
```
using vite seems to work but it also seems to inject itself into every component i use, so when i generate my project, i can see my css is repeated multiple times, some files as much as 300 times. Issue found here on vites repo  https://github.com/vitejs/vite/issues/4448

`css: ["@/assets/styles/main.scss"],`
the above option seems to not do it for every component, but normal scoped sass in .vue files doesnt pick up sass variables and mixins on compilation using this method

**Comment by dsvgl** [comment, marked as answer]
additionalData is only for variables, mixins etc. Do not put actual css-rules in there, or you get what you described!
Eg. having a `_vars.scss` file with `$foo: red;` in it -> using additionalData will allow you to use `$foo` inside your SFC scss-style-blocks:
```vue
<style lang="scss" scoped>
.my-class {
  color: $foo;
}
</style>
```

The `css` key inside your nuxt.config.ts is for actual (global) css rules that get compiled to css.

**Reply by jpengelbrecht** [reply]
but the problem here is that using additionalData makes the app unusable.  200 lines of vars and mixins getting injected once per component will make any more complex page burst at the seems with duplicated css and needlessly bloat your file size

**Reply by dsvgl** [reply]
Not sure if we're on the same page. Maybe you need to provide a reproduction or screenshots.

I created a quick example that shows usage of a main.scss file and a separate _vars.scss file (only) for scss-varaiables: https://stackblitz.com/edit/github-hq9bhi?file=app.vue
main.scss is included via nuxt.configs' `css` key and _vars.scss via additionalData.
I don't see any duplication.

The OP in your linked issue posted a [reproduction](https://github.com/tastytea-github/vite-global-import-issue-reproduction) which shows the mistake I described: additionalData links to main.scss file which conatins **actual** css-rules: https://github.com/tastytea-github/vite-global-import-issue-reproduction/blob/main/src/styles/main.scss

**Reply by jpengelbrecht** [reply]
ah sorry it seems i was a little dense.   i didnt notice that mixins and vars dont translate over with its converted to css.    so you are right,  those items dont get added they are only used when processing the css.  thanks for your insight man!
